src/components/DropdownMenu.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
+import cn from 'classnames';
+import { peopleFromServer } from '../data/people';
+import { Person } from '../types/Person';
+
+type Props = {
+  setSelectedPerson: (person: Person | null) => void;
+};
+
+const DELAY = 1000;
+
+export const DropdownMenu: React.FC<Props> = ({ setSelectedPerson }) => {
+  const [isMenuShown, setIsMenuShown] = useState(false);
```

Comment:
Would be better to rename the state to something more appropriate cause this is not a menu but list of the users